prow/jenkins/jenkins_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jenkins
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+// getRequestedJob is attempting to determine if this is a job-specific
+// request in a pretty hacky way.
+func getRequestedJob(path string) string {
+	parts := strings.Split(path, "/")
+	jobIndex := -1
+	for i, part := range parts {
+		if part == "job" {
+			// This is a job-specific request. Record the index.
+			jobIndex = i + 1
+			break
+		}
+	}
+	// If this is not a job-specific request, fail for now. Eventually we
+	// are going to proxy queue requests.
+	if jobIndex == -1 {
+		return ""
+	}
+	// Sanity check
+	if jobIndex+1 > len(parts) {
+		return ""
+	}
+	return parts[jobIndex]
+}
+
+func testWrapper(t *testing.T, jobs []string, builds map[string][]JenkinsBuild, status *int) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		t.Logf("test request to %q", r.URL.Path)
+		if r.Method != http.MethodGet {
+			t.Errorf("Bad method: %s", r.Method)
+			return
+		}
+		if r.URL.Path == "/queue/api/json" {
+			fmt.Fprint(w, `{"items": []}`)
+			return
+		}
+		var paths []string
```

Comment:
The declaration and instantiation of `paths` can be done once outside the function that is returned, but inside `testWrapper`.

Reply:
done